pandas/_libs/reduction.pyx

```diff
@@ -45,107 +44,72 @@ cdef class Reducer:
         n, k = (<object>arr).shape
 
         if axis == 0:
-            if not arr.flags.f_contiguous:
-                arr = arr.copy('F')
-
             self.nresults = k
             self.chunksize = n
             self.increment = n * arr.dtype.itemsize
         else:
-            if not arr.flags.c_contiguous:
-                arr = arr.copy('C')
-
+            arr = arr.T
             self.nresults = n
             self.chunksize = k
             self.increment = k * arr.dtype.itemsize
 
         self.f = f
         self.arr = arr
         self.labels = labels
-        self.dummy, self.typ, self.index, self.ityp = self._check_dummy(
-            dummy=dummy)
-
-    cdef _check_dummy(self, object dummy=None):
-        cdef:
-            object index = None, typ = None, ityp = None
-
-        if dummy is None:
-            dummy = np.empty(self.chunksize, dtype=self.arr.dtype)
-
-            # our ref is stolen later since we are creating this array
-            # in cython, so increment first
-            Py_INCREF(dummy)
 
-        else:
-
-            # we passed a Series
-            typ = type(dummy)
-            index = dummy.index
-            dummy = dummy.values
+        # TODO: do we still need this?
+        self._check_dummy(dummy=dummy)
```

Comment:
The way this is written dummy never gets used. I'm not sure what the purpose of dummy is to begin with, so maybe we can get rid of this check and its use in the constructor altogether.  maybe @jbrockmendel knows

Reply:
IIUC this exists so that if we are using a Series subclass, that gets retained in the reduction.  If that can be ensured without needing dummy, thatd be great